website/source/docs/builders/cloudstack.html.md

```diff
@@ -0,0 +1,149 @@
+    CloudStack. The builder takes either an ISO or an existing template as it's
+    source, runs any provisioning necessary on the instance after launching it
+    and then creates a new template from that instance.
+layout: docs
+page_title: CloudStack Builder
+...
+
+# CloudStack Builder
+
+Type: `cloudstack`
+
+The `cloudstack` Packer builder is able to create new templates for use with
+[CloudStack](https://cloudstack.apache.org/). The builder takes either an ISO
+or an existing template as it's source, runs any provisioning necessary on the
+instance after launching it and then creates a new template from that instance.
+
+The builder does *not* manage templates. Once a template is created, it is up
+to you to use it or delete it.
+
+## Configuration Reference
+
+There are many configuration options available for the builder. They are
+segmented below into two categories: required and optional parameters. Within
+each category, the available configuration keys are alphabetized.
+
+In addition to the options listed here, a
+[communicator](/docs/templates/communicator.html) can be configured for this
+builder.
+
+### Required:
+
+-   `api_url` (string) - The CloudStack API endpoint we will connect to.
+
+-   `api_key` (string) - The API key used to sign all API requests.
+
+-   `cidr_list` (array) - List of CIDR's that will have access to the new
+    instance. This is needed in order for any provisioners to be able to
+    connect to the instance. Usually this will be the NAT address of your
+    current location. Only required when `use_local_ip_address` is `false`.
+
+-   `instance_name` (string) - The name of the instance. Defaults to a
+    dynamically generated name (using a time ordered UUID).
+
+-   `network` (string) - The name or ID of the network to connect the instance
+    to.
+
+-   `secret_key` (string) - The secret key used to sign all API requests.
+
+-   `service_offering` (string) - The name or ID of the service offering used
+    for the instance.
+
+-   `soure_iso` (string) - The name or ID of an ISO that will be mounted before
+    booting the instance. This option is mutual exclusive with `source_template`.
+
+-   `source_template` (string) - The name or ID of the template used as base
+    template for the instance. This option is mutual explusive with `source_iso`.
+
+-   `template_name` (string) - The name of the new template. Defaults to a
+    dynamically generated name (using a timestamp).
+
+-   `template_display_text` (string) - The display text of the new template.
+    Defaults to the `template_name`.
+
+-   `template_os` (string) - The name or ID of the template OS for the new
+    template that will be created.
+
+-   `zone` (string) - The name or ID of the zone where the instance will be
+    created.
+
+### Optional:
+
+-   `async_timeout` (int) - The time (in seconds) to wait for async calls to
+    finish. Defaults to 1800 seconds.
```

Comment:
It's nicer to use `time.Duration` so users can use "30m" instead of 1800. For a example see [amazon/common/run_config.go](https://github.com/mitchellh/packer/blob/master/builder/amazon/common/run_config.go#L36)